services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSpec.groovy

```diff
@@ -0,0 +1,463 @@
+        and: "The rest fields in the 'rule' section are empty"
+        [srcSwitchValidateInfo, dstSwitchValidateInfo].each {
+            verifyRuleSectionsAreEmpty(it, ["missing", "excess"])
+        }
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        then: "Check that the switch validate request returns empty sections"
+        Wrappers.wait(WAIT_OFFSET) {
+            def srcSwitchValidateInfoAfterDelete = northbound.switchValidate(srcSwitch.dpId)
+            def dstSwitchValidateInfoAfterDelete = northbound.switchValidate(dstSwitch.dpId)
+            [srcSwitchValidateInfoAfterDelete, dstSwitchValidateInfoAfterDelete].each {
+                verifyRuleSectionsAreEmpty(it)
+                verifyMeterSectionsAreEmpty(it)
+            }
+        }
+    }
+
+    def "Switch validation is able to detect meter info into the 'misconfigured' section"() {
+        when: "Create a flow"
+        def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
+        def flow = flowHelper.addFlow(flowHelper.randomFlow(srcSwitch, dstSwitch))
+        def srcSwitchgetCreatedMeterIds = getCreatedMeterIds(srcSwitch.dpId)
+        def dstSwitchgetCreatedMeterIds = getCreatedMeterIds(dstSwitch.dpId)
+
+        and: "Change bandwidth for the created flow directly in DB"
+        def newBandwidth = flow.maximumBandwidth + 100
+        /** at this point meter is set for given flow. Now update flow bandwidth directly via DB,
+         it is done just for moving meter from the 'proper' section into the 'misconfigured'*/
+        database.updateFlowBandwidth(flow.id, newBandwidth)
+        //at this point existing meters do not correspond with the flow
+
+        then: "Meters info is moved into the 'misconfigured' section"
+        def srcSwitchValidateInfo = northbound.switchValidate(srcSwitch.dpId)
+        def dstSwitchValidateInfo = northbound.switchValidate(dstSwitch.dpId)
+        def srcSwitchCreatedCookies = getCookiesWithMeter(srcSwitch.dpId)
+        def dstSwitchCreatedCookies = getCookiesWithMeter(dstSwitch.dpId)
+
+        def createdCookies = srcSwitchCreatedCookies + dstSwitchCreatedCookies
+
+        srcSwitchValidateInfo.meters.misconfigured*.meterId.size() == 1
+        srcSwitchValidateInfo.meters.misconfigured*.meterId.containsAll(srcSwitchgetCreatedMeterIds)
+        dstSwitchValidateInfo.meters.misconfigured*.meterId.size() == 1
+        dstSwitchValidateInfo.meters.misconfigured*.meterId.containsAll(dstSwitchgetCreatedMeterIds)
+
+        srcSwitchValidateInfo.meters.misconfigured*.cookie.containsAll(srcSwitchCreatedCookies)
+        dstSwitchValidateInfo.meters.misconfigured*.cookie.containsAll(dstSwitchCreatedCookies)
+
+        [srcSwitchValidateInfo, dstSwitchValidateInfo].each {
+            it.meters.misconfigured.each {
+                assert it.rate == flow.maximumBandwidth
+                assert it.flowId == flow.id
+                assert ["KBPS", "BURST", "STATS"].containsAll(it.flags)
```

Comment:
`meterId.size()` verifications can go into this loop

Reply:
I don't think so.
I've corrected the verification above.
```
srcSwitchValidateInfo.meters.misconfigured*.meterId.size() == 1  
-->
srcSwitchValidateInfo.meters.misconfigured.meterId.size() == 1
```

Reply:
you can do 
`it.meters.misconfigured*.meterId.size() == 1` once here instead of doing
```
srcSwitchValidateInfo.meters.misconfigured*.meterId.size() == 1 
dstSwitchValidateInfo.meters.misconfigured*.meterId.size() == 1 
```
twice

Reply:
done